File: player/playerAttack.js
import { user } from '../user/user.js';
import { createEnemy } from '../enemy/createEnemy.js';
import { logBox } from '../logBox/logBox.js';
import { logBoxDisplay } from '../logBox/logBoxDisplay.js';
import { userInterface } from '../user/UI.js';
import { enemyAttack } from '../enemy/enemyAttack.js';

export const playerAttack = () => {
    let damage = Math.floor(Math.random() * user.strength);
    createEnemy.health = createEnemy.health - damage;
    logBox.push(`You attacked the enemy. You did ${damage} damage.`);
    userInterface();
    logBoxDisplay();
    setTimeout(enemyAttack, 1000);
};

File: Battle-Jawn/src/main/java/com/battlejawn/Battle/Jawn/PlayerMove/PlayerAttack.java
package main.java.com.battlejawn.Battle.Jawn.PlayerMove;

import main.java.com.battlejawn.Battle.Jawn.StatusAilments.StatusAilments;

public class PlayerAttack {

    public int damage(int strength) {
        return Math.floor(Math.random() * strength);
    }

    public boolean stagger() {
        if (Math.floor(Math.random() * 10) < 3) {
            return true;
        }
        return false;
    }

    public boolean missed() {
        if (Math.floor(Math.random() * 10) < 1) {
            return true;
        }
        return false;
    }

    public boolean criticalHit() {
        if (Math.floor(Math.random() * 100) < 10) {
            return true;
        }
        return false;
    }

}
